File: app/src/main/java/cz/krystofcejchan/upocasi/fragments/recycler_view_layout_data/WeatherRecyclerViewAdapter.java
package cz.krystofcejchan.upocasi.fragments.recycler_view_layout_data;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import cz.krystofcejchan.upocasi.R;

public final class WeatherRecyclerViewAdapter extends RecyclerView.Adapter<WeatherRecyclerViewAdapter.ViewHolder> {
    private final Context context;
    private final List<RowWeatherModule> data;

    public WeatherRecyclerViewAdapter(Context context, List<RowWeatherModule> data) {
        this.context = context;
        this.data = data;
    }

    @NonNull
    @Override
    public WeatherRecyclerViewAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(context);
        View view = layoutInflater.inflate(R.layout.item_layout_in_recyclerview, parent, false);
        return new WeatherRecyclerViewAdapter.ViewHolder(view);
    }


    @Override
    public void onBindViewHolder(@NonNull WeatherRecyclerViewAdapter.ViewHolder holder, int position) {
        holder.key.setText(data.get(position).getTitle());
        holder.value.setText(data.get(position).getValue());
    }

    /**
     * Returns the total number of items in the data set held by the adapter.
     *
     * @return The total number of items in this adapter.
     */
    @Override
    public int getItemCount() {
        return this.data.size();
    }

    protected static class ViewHolder extends RecyclerView.ViewHolder {
        TextView key, value;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            key = itemView.findViewById(R.id.key);
            value = itemView.findViewById(R.id.value);
        }
    }
}

File: app/src/main/java/cz/krystofcejchan/upocasi/fragments/recycler_view_layout_data/WeatherRecyclerViewAdapter.kt
package cz.krystofcejchan.upocasi.fragments.recycler_view_layout_data

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import cz.krystofcejchan.upocasi.R

class WeatherRecyclerViewAdapter(
    private val context: Context?,
    private val data: List<RowWeatherModule>
) : RecyclerView.Adapter<WeatherRecyclerViewAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutInflater = LayoutInflater.from(context)
        val view = layoutInflater.inflate(R.layout.item_layout_in_recyclerview, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.key.text = data[position].title
        holder.value.text = data[position].value
    }

    /**
     * Returns the total number of items in the data set held by the adapter.
     *
     * @return The total number of items in this adapter.
     */
    override fun getItemCount(): Int {
        return data.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var key: TextView
        var value: TextView

        init {
            key = itemView.findViewById(R.id.key)
            value = itemView.findViewById(R.id.value)
        }
    }
}